packages/ibm-products/src/components/Datagrid/types/index.ts

```diff
@@ -0,0 +1,205 @@
+import { CheckboxProps } from '@carbon/react/lib/components/Checkbox';
+import { NumberInputProps } from '@carbon/react/lib/components/NumberInput/NumberInput';
+
+import { JSXElementConstructor, MutableRefObject, ReactNode } from 'react';
+
+type Size = 'xs' | 'sm' | 'md' | 'lg';
+
+export type Filter =
+  | ({
+      column?: string;
+    } & {
+      type: 'date';
+      props: {
+        // DatePickerProps
+        DatePicker?: any;
+        // DatePickerInputProps
+        DatePickerInput?: any;
+      };
+    })
+  | {
+      type: 'number';
+      props: {
+        NumberInput?: NumberInputProps;
+      };
+    }
+  | {
+      type: 'checkbox';
+      props: {
+        FormGroup?: FormGroupProps;
+        Checkbox?: CheckboxProps[];
+      };
+    }
+  | {
+      type: 'radio';
+      props: {
+        FormGroup?: FormGroupProps;
+        RadioButton?: RadioButtonProps[];
+        RadioButtonGroup?: RadioButtonGroupProps;
+      };
+    }
+  | {
+      type: 'multiSelect';
+      props: {
+        MultiSelect?: MultiSelectProps;
+      };
+    };
+
+export interface ReactTableFiltersState {
+  id: string;
+  type: string;
+  value: string;
+}
+
+interface Section {
+  categoryTitle?: string;
+  filters?: Filter[];
+}
+
+export interface FilterFlyoutProps {
+  data?: any;
+  filters?: Filter[];
+  flyoutIconDescription?: string;
+  onFlyoutClose?: () => void;
+  onFlyoutOpen?: () => void;
+  panelIconDescription?: string;
+  primaryActionLabel?: string;
+  reactTableFiltersState?: ReactTableFiltersState[];
+  renderLabel?: () => void;
+  secondaryActionLabel?: string;
+  updateMethod?: string;
+  variation?: string;
+  panelTitle?: string;
+  sections?: Section[];
+  // TODO: Change any
+  autoHideFilters: any;
+}
+
+export interface Column {
+  Header?: string;
+  accessor?: () => void;
+  sticky: 'left' | 'right';
+  id?: string;
+  width?: string;
+  className?: string;
+}
+
+export interface TableProps {
+  className?: string;
+  role?: string;
+  style?: CSSStyleDeclaration;
+}
+
+export interface DataGridState {
+  withVirtualScroll?: boolean;
+  DatagridPagination?: JSXElementConstructor<any>;
+  isFetching?: boolean;
+  tableId?: string;
+  filterProps?: FilterFlyoutProps;
+  className?: string;
+  columns?: Column[];
+  getTableProps?: () => TableProps;
+  getFilterFlyoutProps?: () => FilterFlyoutProps;
+  DatagridActions?: JSXElementConstructor<any>;
+  CustomizeColumnsTearsheet?: JSXElementConstructor<any>;
+  skeletonRowCount?: number;
+  fullHeightDatagrid?: boolean;
+  variableRowHeight?: boolean;
+  useDenseHeader?: boolean;
+  withInlineEdit?: boolean;
+  verticalAlign?: string;
+  gridTitle?: ReactNode;
+  gridDescription?: ReactNode;
+  totalColumnsWidth?: number;
+  gridRef?: MutableRefObject<HTMLDivElement>;
+  setAllFilters?: (...args) => void;
+  getAllFilters?: () => void;
+  DatagridBatchActions?: (args) => ReactNode;
+  batchActions?: boolean;
+  rowSize?: Size;
+  // TODO: Change any
```

Comment:
Much of what is contained in `datagridState` comes from `react-table` `useTable` hook. Can you check if they export any types and extend those?

Reply:
Yes, I've installed the `@types/react-table` package for types and used available types.